internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -661,20 +667,28 @@ func (b *BeatsMonitor) getHttpStreams(
 		"namespace":  "agent",
 		"period":     metricsCollectionIntervalString,
 		"index":      indexName,
-		"processors": processorsForAgentHttpStream(monitoringNamespace, dataset, b.agentInfo),
+		"processors": processorsForAgentHttpStream(agentName, agentName, agentName, monitoringNamespace, dataset, b.agentInfo),
 	}
 	if failureThreshold != nil {
 		agentStream[failureThresholdKey] = *failureThreshold
 	}
 	httpStreams = append(httpStreams, agentStream)
 
+	var edotSubprocessEndpoints []interface{}
+
 	for _, compInfo := range componentInfos {
 		binaryName := compInfo.BinaryName
 		if !isSupportedMetricsBinary(binaryName) {
 			continue
 		}
 
 		endpoints := []interface{}{PrefixedEndpoint(BeatsMonitoringEndpoint(compInfo.ID))}
+		if compInfo.RuntimeManager == component.OtelRuntimeManager && compInfo.ID == httpMonitoringComponentInfoID && b.isOtelRuntimeSubprocess {
```

Comment:
I don't think this condition is completely right. We can have self-monitoring running in beats processes, but other inputs running as beats receivers. With this condition, we wouldn't monitor the otel subprocess, even though we can and should. I think it should be logically `If self-monitoring metrics are enabled, otel is running as a subprocess, and any component uses the otel runtime`. A check similar to what @faec added in https://github.com/elastic/elastic-agent/pull/9928.

Reply:
Valid point, for some reason I thought that we are switch to otel runtime by default for the monitoring needs without supporting beats subprocess, but this was never agreed. And that means that I am gonna have to go down the path of injecting my own monitoring server in edot collector no matter what. thx

Reply:
what if we run edot collector from the pure otel config without any beat receiver?

Reply:
I suppose we could have an otel collector running only hybrid-mode configuration, in which case you do need the monitoring server after all.

Reply:
I am thinking about hybrid-mode, and I believe since we can have otel-specific config in elastic-agent.yml this means that there is a possibility of edot collector running without any beat receiver, right?

Reply:
Yes, that can happen. And it would be a regression, because with embedded mode, we get these metrics as part of the elastic-agent process.